src/main/kotlin/com/hubelias/parkingmeter/port/adapter/rest/WebSecurityConfig.kt

```diff
@@ -0,0 +1,20 @@
+package com.hubelias.parkingmeter.port.adapter.rest
+
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(jsr250Enabled = true)
+private class WebSecurityConfig : WebSecurityConfigurerAdapter() {
+    override fun configure(http: HttpSecurity) {
+        http
+                .authorizeRequests().anyRequest().authenticated()
+                .and().httpBasic()
+                .and().csrf().disable()
```

Comment:
naprawdę potrzebujemy to wyłączyć?

Reply:
Nie przechodziły mi testy endpointów, które wykonywały metody POST lub DELETE. Wyczytałem, że ochrona przed CSRF może być tego powodem oraz, że jej wyłączenie jest bezpieczne w sytuacji gdy aplikacja nie jest używana bezpośrednio z przeglądarki internetowej.

> When you use CSRF protection? Our recommendation is to use CSRF protection for any request that could be processed by a browser by normal users. If you are only creating a service that is used by non-browser clients, you will likely want to disable CSRF protection.
 https://docs.spring.io/spring-security/site/docs/3.2.0.CI-SNAPSHOT/reference/html/csrf.html

No i rzeczywiście testy zaczęły przechodzić. 

Teraz doczytałem jeszcze, że mogę użyć w kodzie testu: `org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf()`. No i wygląda na to, że testy przechodzą wtedy bez powyższej linii. Ogólnie widzę, że muszę jeszcze doczytać co tam się właściwie dzieje i dlaczego :)